crypto/evp/m_sigver.c

```diff
@@ -66,20 +66,36 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
             ctx->pctx = EVP_PKEY_CTX_new_from_pkey(libctx, pkey, props);
         else
             ctx->pctx = EVP_PKEY_CTX_new(pkey, e);
+    } else if (type == NULL && mdname == NULL && pkey == NULL) {
+        reinit = 1;
     }
     if (ctx->pctx == NULL)
         return 0;
 
     locpctx = ctx->pctx;
-    evp_pkey_ctx_free_old_ops(locpctx);
+    if (!reinit)
+        evp_pkey_ctx_free_old_ops(locpctx);
 
     if (props == NULL)
         props = locpctx->propquery;
 
     ERR_set_mark();
 
-    if (evp_pkey_ctx_is_legacy(locpctx))
+    if (evp_pkey_ctx_is_legacy(locpctx)) {
         goto legacy;
+    } else if (reinit) {
+        if (locpctx->operation != (ver ? EVP_PKEY_OP_VERIFYCTX
+                                   : EVP_PKEY_OP_SIGNCTX)
```

Comment:
indentation

Reply:
Personally I would align the `?` and `:`